From b99a6b17e027b65086047f781a9525f7abc745e5 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 20 May 2020 08:56:53 +0300
Subject: [PATCH] [SW-0] goya2: arcboot: ram: fix compilation error

Due to update to goya2 specs, cpu pll header files
and corresponding register names have been removed.
This patch fixes resulting compilation errors
by removing complete arcboot ram functionality.
Infrastructure is kept as it is for future use.

Change-Id: Ie0df39f21445aed1fcd4076a8ac27053aab3dd7f
---

diff --git a/app/arcboot_ram.c b/app/arcboot_ram.c
index 507ac6b..36b421d 100644
--- a/app/arcboot_ram.c
+++ b/app/arcboot_ram.c
@@ -5,99 +5,7 @@
  * SPDX-License-Identifier:	GPL-2.0+
  */
 
-#include "arcboot.h"
-#include "delay.h"
-
-static void init_misc(void)
-{
-	/* Enable gpio 13/12 as outputs used by LEDs */
-	regmap_writel(0x3000, mmPSOC_GPIO0_OUTENSET);
-	/* Trigger boot manager on next reset */
-	regmap_writel(0x1, mmPSOC_GLOBAL_CONF_BOOT_SEQ_RE_START);
-	/* Set status to CPU_BOOT_STATUS_IN_BTL */
-	regmap_writel(0x4, mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS);
-}
-
-static void init_pll_cpu(void)
-{
-	/* Gate the clock */
-	regmap_writel(0, mmPSOC_CPU_PLL_DIV_0_RLX_CLK_GATER);
-	regmap_writel(0, mmPSOC_CPU_PLL_DIV_1_RLX_CLK_GATER);
-	regmap_writel(0, mmPSOC_CPU_PLL_DIV_2_RLX_CLK_GATER);
-	regmap_writel(0, mmPSOC_CPU_PLL_DIV_3_RLX_CLK_GATER);
-	/* Actual values written into the register are (val - 1) */
-	regmap_writel(0x503b, mmPSOC_CPU_PLL_CTRL_CFG_DIV); /* 1, 6, 60 */
-	regmap_writel(0x11, mmPSOC_CPU_PLL_CTRL_DATA_CHNG);
-
-	/* use PLL clock, no divisor */
-	regmap_writel(0x1, mmPSOC_CPU_PLL_CTRL_DIV_SEL_0);
-
-	/* use PLL clock, no divisor */
-	regmap_writel(0x1, mmPSOC_CPU_PLL_CTRL_DIV_SEL_1);
-
-	regmap_writel(0x4, mmPSOC_CPU_PLL_CTRL_DIV_FACTOR_2);	/* 5 */
-	regmap_writel(0x1, mmPSOC_CPU_PLL_CTRL_DIV_FACTOR_CMD_2);
-	regmap_writel(0x1, mmPSOC_CPU_PLL_CTRL_DIV_EN_2);
-	/* use PLL clock, w/ divisor */
-	regmap_writel(0x3, mmPSOC_CPU_PLL_CTRL_DIV_SEL_2);
-
-	/* use REF clock, w/o divisor */
-	regmap_writel(0x0, mmPSOC_CPU_PLL_CTRL_DIV_SEL_3);
-
-	/* Wait 40us before un-gating the clock */
-	udelay(40);
-	regmap_writel(1, mmPSOC_CPU_PLL_DIV_0_RLX_CLK_GATER);
-	regmap_writel(1, mmPSOC_CPU_PLL_DIV_1_RLX_CLK_GATER);
-	regmap_writel(1, mmPSOC_CPU_PLL_DIV_2_RLX_CLK_GATER);
-	regmap_writel(1, mmPSOC_CPU_PLL_DIV_3_RLX_CLK_GATER);
-}
-
-/* Functionality used to be run by BTL */
-static void arm_prepare(void)
-{
-	init_misc();
-	init_pll_cpu();
-}
-
-static void arm_wakeup(void)
-{
-	/*
-	 * TODO: set proper value '1'|'2'
-	 * according to what pass verification
-	 * '1' - Primary SPI image [Default]
-	 * '2' - Secondary SPI image
-	 * In the meantime we used predefined '0'
-	 */
-	unsigned int arcboot_id = BTLID;
-
-	/*
-	 * Traditionally we used scratchpad registers for
-	 * passing the current running BTL ID and VER to CPU SW
-	 * TODO: Need do it in case of ARCBOOT
-	 */
-	regmap_writel(arcboot_id, mmPSOC_GLOBAL_CONF_SCRATCHPAD_31);
-
-	/* Passing the BTL version to preboot through the scratchpad 28 register */
-	regmap_writel(BTLVER, mmPSOC_GLOBAL_CONF_SCRATCHPAD_28);
-
-	/* Set CPU0 Reset address to PREBOOT */
-	regmap_writel(PREBOOT_ADDR_L, mmCPU_CA53_CFG_RST_ADDR_LSB_0);
-	regmap_writel(PREBOOT_ADDR_H, mmCPU_CA53_CFG_RST_ADDR_MSB_0);
-
-	/* Wake up CPU0, after we put all CPUs in reset */
-	regmap_writel(0x0, mmCPU_CA53_CFG_ARM_RST_CONTROL);
-	regmap_writel(0x10111, mmCPU_CA53_CFG_ARM_RST_CONTROL);
-}
-
-static void arm_call(void)
-{
-	arm_prepare();
-	arm_wakeup();
-}
-
 int arcboot_ram_main(void)
 {
-	arm_call();
-
 	return 0;
 }
